packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart

```diff
@@ -0,0 +1,632 @@
+      _trySetUserAgent(settings.userAgent),
+      _trySetZoomEnabled(settings.zoomEnabled),
+    ]);
+  }
+
+  @override
+  Future<String> evaluateJavascript(String javascript) async {
+    return runJavascriptReturningResult(javascript);
+  }
+
+  @override
+  Future<void> runJavascript(String javascript) async {
+    await webView.evaluateJavascript(javascript);
+  }
+
+  @override
+  Future<String> runJavascriptReturningResult(String javascript) async {
+    return await webView.evaluateJavascript(javascript) ?? '';
+  }
+
+  @override
+  Future<void> addJavascriptChannels(Set<String> javascriptChannelNames) {
+    return Future.wait(
+      javascriptChannelNames.where(
+        (String channelName) {
+          return !_javaScriptChannels.containsKey(channelName);
+        },
+      ).map<Future<void>>(
+        (String channelName) {
+          final WebViewAndroidJavaScriptChannel javaScriptChannel =
+              WebViewAndroidJavaScriptChannel(
+                  channelName, javascriptChannelRegistry);
+          _javaScriptChannels[channelName] = javaScriptChannel;
+          return webView.addJavaScriptChannel(javaScriptChannel);
+        },
+      ),
+    );
+  }
+
+  @override
+  Future<void> removeJavascriptChannels(
+    Set<String> javascriptChannelNames,
+  ) {
+    return Future.wait(
+      javascriptChannelNames.where(
+        (String channelName) {
+          return _javaScriptChannels.containsKey(channelName);
+        },
+      ).map<Future<void>>(
+        (String channelName) {
+          final WebViewAndroidJavaScriptChannel javaScriptChannel =
+              _javaScriptChannels[channelName]!;
+          _javaScriptChannels.remove(channelName);
+          return webView.removeJavaScriptChannel(javaScriptChannel);
+        },
+      ),
+    );
+  }
+
+  @override
+  Future<String?> getTitle() => webView.getTitle();
+
+  @override
+  Future<void> scrollTo(int x, int y) => webView.scrollTo(x, y);
+
+  @override
+  Future<void> scrollBy(int x, int y) => webView.scrollBy(x, y);
+
+  @override
+  Future<int> getScrollX() => webView.getScrollX();
+
+  @override
+  Future<int> getScrollY() => webView.getScrollY();
+
+  Future<void> _release() => webView.release();
+
+  void _setCreationParams(CreationParams creationParams) {
+    final WebSettings? webSettings = creationParams.webSettings;
+    if (webSettings != null) {
+      updateSettings(webSettings);
+    }
+
+    final String? userAgent = creationParams.userAgent;
+    if (userAgent != null) {
+      webView.settings.setUserAgentString(userAgent);
+    }
+
+    final AutoMediaPlaybackPolicy autoMediaPlaybackPolicy =
+        creationParams.autoMediaPlaybackPolicy;
+    switch (autoMediaPlaybackPolicy) {
+      case AutoMediaPlaybackPolicy.always_allow:
+        webView.settings.setMediaPlaybackRequiresUserGesture(false);
+        break;
+      default:
+        webView.settings.setMediaPlaybackRequiresUserGesture(true);
+    }
```

Comment:
Why not just:
`webView.settings.setMediaPlaybackRequiresUserGesture(autoMediaPlaybackPolicy != AutoMediaPlaybackPolicy.always_allow)`?

Reply:
I prefer to use `switch` statements for enums because the Dart code will throw a lint error that one of the values wasn't listed. We had a bug once when we didn't support all the enum values.

Reply:
`default` covers all cases, so you'll never get a linter warning here no matter what is added to the enum.

> We had a bug once when we didn't support all the enum values.

If someone is adding new functionality, they should be adding test coverage, which is how we should catch this. Relying on the lint (by removing `default` to enable it) would make adding new enum values repository-breaking. We've already had a sever issue with this; see https://github.com/flutter/flutter/issues/89866

Reply:
I don't think flutter/flutter#89866 actually provides a recommendation on the implementation. Are you suggesting that we keep the `default` value? If the platform interface adds a new `enum` value, wouldn't only an integration test in the main plugin (e.g. `webview_flutter`) catch a change like that? The unit tests in the platform interface and main plugin should be platform agnostic. 

I think I prefer option 3 from flutter/flutter#89866. That way there is a process to locate everywhere enum used in the implementations. And we can add `TODO`s to update at a later point or in a followup PR.

Reply:
> I don't think [flutter/flutter#89866](https://github.com/flutter/flutter/issues/89866) actually provides a recommendation on the implementation. Are you suggesting that we keep the `default` value?

I think in cases where we can have a reasonable `default`, we should do that, because it reduces the chances of cross-package breakage. (We don't yet have a way to detect that in CI.)

> If the platform interface adds a new `enum` value, wouldn't only an integration test in the main plugin (e.g. `webview_flutter`) catch a change like that? The unit tests in the platform interface and main plugin should be platform agnostic.

I'm not sure I follow. If someone is adding a new enum value, it's presumably to add new functionality based on that new value. That would generally require a change to each platform implementation package to actually use the new enum value, and those changes should have tests. So it should be something that would be caught by tests both at the platform implementation level and potentially at the app-facing level.

Reply:
Ah, ok. I assumed someone would be able to make a change to a platform interface `enum` without needing to update the implementations. If that's the case, then yea we should be able to catch all the uses if we require this.

Reply:
I changed it to your suggestion